src/Protocol/Http/Authenticator/JwtAuthenticator.php

```diff
@@ -30,24 +30,61 @@ class JwtAuthenticator implements AuthenticatorInterface
      */
     private $jwt;
 
+    /**
+     * @var string
+     */
+    private $jws = '';
+
+    /**
+     * @var \DateInterval
+     */
+    private $jwsLifetime = null;
+
+    /**
+     * @var \DateTime
+     */
+    private $jwsValidTo = null;
+
     /**
      * Constructor.
      *
-     * @param JwtInterface $jwt
+     * @param JwtInterface  $jwt
+     * @param \DateInterval $jwsLifetime
      */
-    public function __construct(JwtInterface $jwt)
+    public function __construct(JwtInterface $jwt, \DateInterval $jwsLifetime = null)
     {
         $this->jwt = $jwt;
+
+        $this->jwsLifetime = is_null($jwsLifetime) ? new \DateInterval('P1H') : $jwsLifetime;
+        if ($jwsLifetime->invert) {
+            throw new \InvalidArgumentException('JWS lifetime must not be inverted');
```

Comment:
Good solution! But please add the `@throw` tag with info to doc comment of the method.